src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -251,11 +253,12 @@ private class MergedBatchPutCommand : TransactionOperationsMerger.MergedTransact
             public bool IsDisposed => _isDisposed;
 
             private readonly DocumentsOperationContext _context;
+            private const string PreV4RevisionsDocumentKey = "/revisions/";
 
             public MergedBatchPutCommand(DocumentDatabase database, long buildVersion, Logger log)
             {
                 _database = database;
-                _buildVersion = buildVersion;
+                _isPreV4Build = BuildVersion.IsPreV4(buildVersion);
```

Comment:
Maybe wrap build version in sth like this
```
public class BuildVersion
{
    public class BuildVersion(int buildNumber)
    {
        Type = <calculate here>;
        Number = buildNumber;
    }

    public enum BuildVersionType
    {
        V3,
        V4,
        etc
    }
}
```
pass it through the code and calculate everything just once?